src/hooks/hookResources.ts

```diff
@@ -503,6 +503,154 @@ export function handleHook(
   }
 }
 
+// process the MedicationRequests to add the Medication into contained resources
+function processMedicationRequests(medicationRequestsBundle: Bundle) {
+  medicationRequestsBundle?.entry?.forEach(entry => {
+    if (entry?.resource?.resourceType === 'MedicationRequest') {
+      if (entry?.resource?.medicationReference) {
+        const medicationReference = entry?.resource?.medicationReference;
+        medicationRequestsBundle?.entry?.forEach(e => {
+          if (e?.resource?.resourceType === 'Medication') {
+            if (
+              e?.resource?.resourceType + '/' + e?.resource?.id ===
+              medicationReference?.reference
+            ) {
+              if (entry) {
+                if (entry.resource) {
+                  const reference = e?.resource;
+                  const request = entry.resource as MedicationRequest;
+
+                  // add the reference as a contained resource to the request
+                  if (!request?.contained) {
+                    request.contained = [];
+                    request.contained.push(reference);
+                  } else {
+                    // only add to contained if not already in there
+                    let found = false;
+                    request.contained.forEach(c => {
+                      if (c.id === reference.id) {
+                        found = true;
+                      }
+                    });
+                    if (!found) {
+                      request.contained.push(reference);
+                    }
+                  }
+                }
+              }
+            }
+          }
+        });
+      }
+    }
+  });
+}
```

Comment:
```suggestion
// process the MedicationRequests to add the Medication into contained resources
const refersToMedication = (entry: BundleEntry<FhirResource>): boolean =>
  entry.resource?.resourceType === 'Medication';

const refersToMedicationRequest = (entry: BundleEntry<FhirResource>): boolean =>
  entry.resource?.resourceType === 'MedicationRequest';

const refersToMedicationWithMedicationReference = (e: BundleEntry<MedicationRequest>): boolean =>
  !!e.resource?.medicationReference;

const isBundleEntryMedicationReferenced =
  (medicationRequestEntry: BundleEntry<MedicationRequest>) =>
  (medicationEntry: BundleEntry<Medication>): boolean =>
    medicationEntry?.resource?.resourceType + '/' + medicationEntry?.resource?.id ===
    medicationRequestEntry.resource?.medicationReference?.reference;

const createBundleEntryWhoseMedicationRequestContainsReferencedMedication =
  (medicationEntries: BundleEntry<Medication>[]) =>
  (medicationRequestEntry: BundleEntry<MedicationRequest>): BundleEntry<MedicationRequest> => {
    if (!medicationRequestEntry.resource) {
      return medicationRequestEntry;
    }

    const referencedMedication: Medication = medicationEntries.find(
      isBundleEntryMedicationReferenced(medicationRequestEntry)
    )?.resource!;

    const contained = getContained(medicationRequestEntry, referencedMedication);

    const mutatedMedicationRequestEntry: BundleEntry<MedicationRequest> = {
      ...medicationRequestEntry,
      resource: {
        ...medicationRequestEntry.resource,
        contained
      }
    };

    return mutatedMedicationRequestEntry;
  };

function getContained(
  medicationRequestEntry: BundleEntry<MedicationRequest>,
  referencedMedication: Medication
) {
  const existingContained = medicationRequestEntry.resource?.contained;
  if (existingContained) {
    const foundReferencedMedication = existingContained.find(
      c => c.id === referencedMedication?.id
    );
    if (foundReferencedMedication || !referencedMedication) {
      return existingContained;
    }
    return [...existingContained, referencedMedication];
  }
  return [referencedMedication];
}

function processMedicationRequests(
  medicationRequestsBundle: Bundle<MedicationRequest | Medication | FhirResource> | undefined
): Bundle<MedicationRequest | Medication | FhirResource> | undefined {
  if (!medicationRequestsBundle) {
    return undefined;
  }

  const { entry = [], ...rest } = medicationRequestsBundle;

  const medicationRequestEntries = entry.filter(
    refersToMedicationRequest
  ) as BundleEntry<MedicationRequest>[];

  const medicationRequestEntriesWithMedicationReference = medicationRequestEntries.filter(
    refersToMedicationWithMedicationReference
  );

  const medicationEntries = entry.filter(refersToMedication) as BundleEntry<Medication>[];

  const medicationRequestEntriesMutatedWithMedicationReference =
    medicationRequestEntriesWithMedicationReference.map(
      createBundleEntryWhoseMedicationRequestContainsReferencedMedication(medicationEntries)
    );

  const otherEntries = entry.filter(e => !refersToMedication(e) && !refersToMedicationRequest(e));

  const medicationRequestEntriesWithoutMedicationReference = medicationRequestEntries.filter(
    e => !refersToMedicationWithMedicationReference(e)
  );

  return {
    ...rest,
    entry: [
      ...otherEntries,
      ...medicationEntries,
      ...medicationRequestEntriesWithoutMedicationReference,
      ...medicationRequestEntriesMutatedWithMedicationReference
    ]
  };
}
```

Reply:
This is a good suggestion but maybe should be done as another ticket. This ticket simply copied the code from one place to another and left it unchanged. Can you write this new ticket and include this suggestion, or a link to it in the new ticket?

Reply:
I can do that yes